azure-quantum/azure/quantum/target/target.py

```diff
@@ -2,19 +2,21 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 ##
-from typing import TYPE_CHECKING, Any, Dict
+from typing import TYPE_CHECKING, Any, Dict, Union, Optional
+from abc import ABC
 import io
 import json
-from typing import Any, Dict
 
-from azure.quantum._client.models import TargetStatus
+from azure.quantum._client.models import TargetStatus, SessionDetails
+from azure.quantum._client.models._enums import SessionJobFailurePolicy
 from azure.quantum.job.job import Job
+from azure.quantum.job.session import Session, SessionHost
 from azure.quantum.job.base_job import ContentType
+
 if TYPE_CHECKING:
     from azure.quantum import Workspace
 
-
-class Target:
+class Target(ABC, SessionHost):
```

Comment:
What is this ABC?

Reply:
When I was learning Python I had exactly the same question :-)
https://docs.python.org/3/library/abc.html